lib/langchain/tool_definition.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+require "json"
+
+#
+# Extends a class to be used as a tool in the assistant.
+# A tool is a collection of actions (methods) used to perform specific tasks.
+#
+# == Usage
+#
+# 1. Extend your class with {Langchain::ToolDefinition}
+# 2. Use {#define_action} to define each action of the tool
```

Comment:
I'm wondering if we should call this method `define_function` or `define_tool_function`. What do you think?